Solutions/Ais.Net.Receiver/StorageClient.cs

```diff
@@ -15,33 +16,58 @@ namespace Endjin.Ais.Receiver
     using System.Text;
     using System.Threading.Tasks;
 
-    public class StorageClient
+    public class StorageClient : IStorageClient
     {
-        private AppendBlobClient container;
-        private IConfiguration configuration;
-        private Stream stream;
+        private readonly IConfiguration configuration;
+        private AppendBlobClient appendBlobClient;
+        private BlobContainerClient blobContainerClient;
 
         public StorageClient(IConfiguration configuration)
         {
             this.configuration = configuration;
         }
 
-        public async Task AppendMessages(IEnumerable<string> messages)
+        public async Task PersistAsync(IEnumerable<string> messages)
         {
-            await this.InitialiseConnectionAsync().ConfigureAwait(false);
+            await this.InitialiseContainerAsync().ConfigureAwait(false);
+            await using MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(messages.Aggregate(new StringBuilder(), (sb, a) => sb.AppendLine(string.Join(",", a)), sb => sb.ToString())));
+            await this.appendBlobClient.AppendBlockAsync(stream).ConfigureAwait(false);
+        }
 
-            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(messages.Aggregate(new StringBuilder(), (sb, a) => sb.AppendLine(String.Join(",", a)), sb => sb.ToString())))) 
+        private async Task InitialiseContainerAsync()
+        {
+            try
+            {
+                this.blobContainerClient = new BlobContainerClient(
+                    this.configuration[ConfigurationKeys.ConnectionString],
+                    this.configuration[ConfigurationKeys.ContainerName]);
+                this.appendBlobClient = new AppendBlobClient(
+                    this.configuration[ConfigurationKeys.ConnectionString],
+                    this.configuration[ConfigurationKeys.ContainerName],
+                    $"raw/{DateTimeOffset.Now:yyyy}/{DateTimeOffset.Now:MM}/{DateTimeOffset.Now:dd}/{DateTimeOffset.Now:yyyyMMddTHH}.nm4");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
```

Comment:
I'm guessing this was here because you noticed that you were never actually seeing the errors. But it was only necessary because of the abuse of the `Select` operator earlier on. If you change to an `await foreach` to write out the buffers, the exception should bubble out in to that code properly.